Update to work with Vue 3.

hsorby opened the pull request:
I have made changes so that I can use this with Vue 3. I hope you don't mind but I published a namespaced version ([@hsorby/vue3-katex](https://www.npmjs.com/package/@hsorby/vue3-katex)) so that I can use it now.  Happy to make further changes to this PR to bring it more inline with existing codebase, otherwise it will serve as a place for people to (maybe) find out that there is a Vue 3 option available.

hsorby committed:
Convert to work with Vue 3. Tests not updated.

hsorby committed:
Progress towards working with Vue 3.

hsorby committed:
Update to work with Vue 3.

hsorby committed:
Update README.md for Vue 3.

hsorby committed:
Remove the use of VDomPurifyHtml.  Leave comment for possible inclusion.

hsorby committed:
0.6.1-0

hsorby committed:
0.6.2-0

hsorby committed:
Try to improve packaging information.

hsorby committed:
Inject global options using reference variable.

hsorby committed:
Update test snapshots.

hsorby committed:
Set version to 0.6.0-rc.2.

hsorby committed:
Bump version to 0.6.0-rc.3.

Commit:
Updated Katex and Vue3 versions
* Using npm 16.14

Commit:
Adding typescript module declaration stub

hsorby committed:
Progress on package creation.

hsorby committed:
Update katex, set version to 0.6.0-rc.4.

hsorby committed:
Update to latest snapshot, merge in changes from origin/main fixing conflicts.

tobiasdiez commented:
@lucpotage would be nice if this PR could be merged!

lucpotage commented:
Hi @tobiasdiez,
Thanks for the PR, I spent some time to read it and I'm almost ready to merge it.
Could you please keep the original name (vue-katex), repository URL, and add your name in the list of contributors in package.json?

tobiasdiez commented:
Nice! Btw the PR the nice work of @hsorby, I'm just a bystander that would like to see the vue 3 support ;-)

hsorby committed:
Revert @hsorby/vue3-katex to vue-katex in README.md

hsorby committed:
Fix more @hsorby/vue3-katex in README.md

hsorby committed:
Revert hsorby locations for lucpotage in package.json

hsorby committed:
Set package name to namespace @hsorby/vue3-katex.

hsorby committed:
Remove yarn.lock.

hsorby committed:
Merge in changes from gatlinnewhouse/main into gatlinnewhouse/main fixing conflicts.

hsorby committed:
Fix syntax error in package.json.

hsorby committed:
Update snapshots.

hsorby committed:
Turn off vue/no-v-text-v-html-on-component eslint error.

hsorby committed:
Update depedencies and devDependencies.

hsorby committed:
Set vue-dompurify-html as a peer depenency as it is not actually used.

hsorby committed:
Uninstall vue-dompurify-html.

hsorby committed:
Update test snapshots.

hsorby committed:
Set type: module in package.json.

Visgaard commented:
I'm also just a bystander looking to use KaTeX with Vue 3. With

`npm i @hsorby/vue3-katex katex -P`

and 
`import katex from 'katex'
import 'katex/dist/katex.min.css'
import VueKatex from 'vue-katex'

const app = createApp(App)
app.use(router)
app.use(katex)
app.use(VueKatex)`

I'm getting errors:
Module not found: Error: Can't resolve 'vue-katex'

hsorby committed:
Update package.json

Correct name of package.

aguirremac commented:
Hi @hsorby, I'm using your unmerge PR (@hsorby/vue3-katex). Do you know how can we add extensions for this package like mhchem support?

I tried adding the cdn as stated here: https://github.com/KaTeX/KaTeX/blob/main/contrib/mhchem/README.md
but it doesn't seem to work.

hsorby committed:
Add options to enable the use of contrib content from katex.

hsorby committed:
Set version to 0.6.0-rc.6.

hsorby committed:
Reset package name to proper package name.

hsorby commented:
> Hi @hsorby, I'm using your unmerge PR (@hsorby/vue3-katex). Do you know how can we add extensions for this package like mhchem support?
> 

I have added an option to the package to enable `mhchem` support, you will need to upgrade to `0.6.0-rc.6` for this though.
When you `use` the package add the option `{ mhchem: true }` and you should have `mhchem` support. I have only given this the bare minimum of testing and it worked for me (tm). If it doesn't work for you show us the error and I'll have a go at fixing it. Check the README I added some documentation there.

aguirremac commented:
Hi @hsorby, thank you for your response. Unfortunately, it's still not working for me.

Here's my plugin btw. I'm using it in Nuxt3.

```
import 'katex/dist/katex.min.css'

import VueKatex from '@hsorby/vue3-katex'
import { defineNuxtPlugin } from '#app'

export default defineNuxtPlugin((nuxtApp) => {
  nuxtApp.vueApp.use(VueKatex, {
    mhchem: true,
    globalOptions: {
      delimiters: [
        { left: '$$', right: '$$', display: true },
        { left: '$', right: '$', display: false },
        { left: '\begin{equation}', right: 'end{equation}', display: true },
        { left: '\begin{align}', right: 'end{align}', display: true }
      ]
    }
  })
})

```
Here's the error on console.

> chunk-4OF65L5M.js:1565 KaTeX auto-render: Failed to parse `\ce{NaCl, KBr}` with  ParseError: KaTeX parse error: Undefined control sequence: \ce at position 1: \̲c̲e̲{NaCl, KBr}
>     at new ParseError2 (katex.js:107:14)
>     at Parser2.parseGroup (katex.js:18346:17)
>     at Parser2.parseAtom (katex.js:17781:21)
>     at Parser2.parseExpression (katex.js:17643:23)
>     at Parser2.parse (katex.js:17575:24)
>     at parseTree2 (katex.js:18583:21)

hsorby commented:
I tried a simple example with Nuxt3 (available [here](https://github.com/hsorby/test-nuxt3-vue3-katex)). And I got this result:
<img width="431" alt="image" src="https://github.com/lucpotage/vue-katex/assets/778048/583e6514-3931-440c-a03f-2d3627d821a1">

I would need to see more of your setup to get any further with your particular issue. I did release another version `0.6.0-rc.7` to clear an error when loading the mchem module, I assume there was some searching of an appropriate library happening in the background. `0.6.0-rc.6` did work for me but an error was reported in the console which I think `0.6.0-rc.7` fixes.

aguirremac commented:
I was actually using the v-katex:auto to render the included LaTeX or mhchem characters from an API. My pages goes like this.

`<template>
  <div v-katex:auto v-html="data.body"></div>
</template>`

Note: the data.body is being fetched from an API.
The LaTeX characters are rendering fine, but not the mhchem characters.

aguirremac commented:
Hi @hsorby , if I do this on my app. It's working v-katex="'\\ce{CO2 + C -> 2 CO}'". The problem is I'm rendering an HTML data from an API with included mchem/latex characters using v-html. It doesn't auto render it using v-katex:auto unlike for the math characters.

tobiasdiez commented:
I guess by now it's clear that this package is no longer actively maintained. @hsorby what do you think about making your fork the "official" successor, i.e. setup a github repo, change package metadata etc?

hsorby commented:
> I guess by now it's clear that this package is no longer actively maintained. @hsorby what do you think about making your fork the "official" successor, i.e. setup a github repo, change package metadata etc?

I have started setting up a repository here: [vue3-plugins/vue3-katex](https://github.com/vue3-plugins/vue3-katex). Feel free to ask if you want to be added to the team for this project.

hsorby committed:
Merge pull request #1 from hsorby/main

Moving towards vue3-katex

hsorby committed:
Move to mjs modules.

hsorby committed:
Merge branch 'main' of github.com:vue3-plugins/vue3-katex

hsorby committed:
Upgrade packages/.

hsorby committed:
Update packages to latest version. Switch from using Jest to vitest.

hsorby committed:
Improve output of package built with rollup.

hsorby committed:
Update location of package to new home.

hsorby committed:
Add GitHub action to lint and test on pull request.

hsorby committed:
Change 'prepublish' script to 'prerelease'.

hsorby committed:
Merge branch 'main' of github.com:vue3-plugins/vue3-katex

hsorby committed:
Update README.md.

hsorby committed:
Set a GitHub build status badge in the README.

hsorby committed:
Merge pull request #2 from hsorby/main

Update packages and set name of package to vue3-katex

hsorby committed:
Add release status check on push to main branch.

hsorby committed:
Add release status check on push to main branch.

hsorby committed:
Merge branch 'main' of github.com:vue3-plugins/vue3-katex

hsorby committed:
Add test coverage script for coverage help.

hsorby committed:
Add @vitest/coverage-v8 dev package.

hsorby committed:
Add code coverage and upload to codecov.

hsorby committed:
Run coverage.

hsorby committed:
Set codecov branch to main.

lucpotage commented:
@hsorby Thanks for your invite. I honestly don’t have the time to manage this plugin anymore. I added you on npmjs so that you can publish with` vue-katex` handle. Thanks also for your patience! Feel free to reach at me if I need to do extra stuff to make it work for you.

I’m gonna archive this repository.

hsorby committed:
0.7.1-0

hsorby committed:
Reset package.json version number.

hsorby committed:
0.7.1-0

hsorby committed:
0.7.2-0

hsorby committed:
0.7.2-1

hsorby committed:
0.7.2-2

hsorby committed:
0.7.2

hsorby committed:
Reset package.json version number.

hsorby committed:
0.7.1-0

hsorby committed:
Set version to 0.7.0-0.

hsorby committed:
Add publish workflow on version tags.

hsorby committed:
0.7.0-1

hsorby committed:
Set actions/setup-node to use version 4.

hsorby committed:
Tidy publish.yml.

hsorby committed:
Import katex contrib packages always, dynamic import does not seem to work.

hsorby committed:
Try and fix up props table in README.

hsorby committed:
0.7.0-2

hsorby committed:
Update README with reference to example website.

hsorby committed:
0.7.0-3

hsorby committed:
Update README.

hsorby committed:
0.7.0

hsorby committed:
Update packages.

hsorby committed:
Update publish script.

hsorby committed:
Tidy up comments.

hsorby committed:
Merge branch 'main' into main

hsorby committed:
Swap out 18.x for 24.x.

hsorby committed:
Merge branch 'main' of github.com:hsorby/vue3-katex

hsorby committed:
Merge pull request #3 from hsorby/main

Update to latest packages

hsorby committed:
Fix type in yarn prerelease command.

hsorby committed:
Set version to 0.8.0-2.

hsorby committed:
Handle prerelease tags in publish workflow.

hsorby committed:
Set version to 0.8.0.

hsorby committed:
Update dependencies for package.
